wgpu-core/src/instance.rs

```diff
@@ -71,8 +71,24 @@ pub struct Instance {
 
 impl Instance {
     pub fn new(name: &str, instance_desc: wgt::InstanceDescriptor) -> Self {
+        fn env(key: &str) -> Option<bool> {
+            std::env::var(key).ok().map(|s| {
+                match s.as_str() {
+                    "0" => false,
+                    _ => true,
+                }
+            })
+        }
         fn init<A: HalApi>(_: A, instance_desc: &wgt::InstanceDescriptor) -> Option<A::Instance> {
             if instance_desc.backends.contains(A::VARIANT.into()) {
+                let mut flags = instance_desc.flags;
+                if let Some(bit) = env("WGPU_VALIDATION") {
+                    flags.set(wgt::InstanceFlags::VALIDATION, bit);
+                }
+                if let Some(bit) = env("WGPU_DEBUG") {
+                    flags.set(wgt::InstanceFlags::DEBUG, bit);
+                }
+
```

Comment:
I don't think this is a good idea, wgpu is a library and we shouldn't become sensitive to env variables.
We can however add an util function similar to the ones we already have that read and set these based on env variables.

Reply:
I was hesitating between this and adding `InstanceFlags::from_env()`, I'll switch over to that.